packages/camera/camera_android_camerax/lib/src/rotated_preview.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+/// Widget that rotates the camera preview to be upright according to the
+/// current device orientation.
```

Comment:
Nit: Is this actually based the device orientation or the "app" orientation. 
I think it is based on the apps orientation. This is especially important as newer versions of android allow forcing the orientation of apps independent of the sensor.